src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -543,29 +589,285 @@ TEST_CASE("Struct class type metadata")
+            }
+            enum E1 {}  
+        }
+    )" };
+    std::string assembly_name = "testidl";
+    xlang::meta::reader::database db{ run_and_save_to_memory(test_idl, assembly_name) };
+
+    REQUIRE(db.TypeRef.size() == TYPE_REF_OFFSET + 3);
+    auto const& E1_ref = find_type_by_name<TypeRef>(db.TypeRef, "E1", "N");
+    auto const& ICombo_ref = find_type_by_name<TypeRef>(db.TypeRef, "IComboBox", "N");
+    auto const& S1_ref = find_type_by_name<TypeRef>(db.TypeRef, "S1", "N");
+    REQUIRE(E1_ref.TypeName() == "E1");
+    REQUIRE(ICombo_ref.TypeName() == "IComboBox");
+    REQUIRE(S1_ref.TypeName() == "S1");
+
+    REQUIRE(db.TypeDef.size() == TYPE_DEF_OFFSET + 3);
+    REQUIRE(db.TypeDef[TYPE_DEF_OFFSET + 1].TypeName() == "IComboBox");
+    auto const& combo = db.TypeDef[TYPE_DEF_OFFSET + 1];
+    test_interface_type_properties(combo);
+
+    REQUIRE(size(combo.MethodList()) == 1);
+    auto const& method_list = combo.MethodList();
+
+    REQUIRE(method_list.first[0].Name() == "Draw");
+    REQUIRE(method_list.first[0].Flags().value == interface_method_attributes().value);
+    {
+        auto const& Draw_sig = method_list.first[0].Signature();
+        REQUIRE(std::get<coded_index<TypeDefOrRef>>(Draw_sig.ReturnType().Type().Type()).TypeRef() == S1_ref);
+        auto const& Draw_param_sig = Draw_sig.Params().first[0];
+        REQUIRE(std::get<coded_index<TypeDefOrRef>>(Draw_param_sig.Type().Type()).TypeRef() == E1_ref);
+    }
+}
+
+TEST_CASE("Interface property metadata")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            struct S1 {}    
+            interface IComboBox 
+            {
+                S1 property1;
+            }
+            enum E1 {}  
+        }
+    )" };
+    std::string assembly_name = "testidl";
+    xlang::meta::reader::database db{ run_and_save_to_memory(test_idl, assembly_name) };
+
+    REQUIRE(db.TypeRef.size() == TYPE_REF_OFFSET + 3);
+    find_type_by_name<TypeRef>(db.TypeRef, "IComboBox", "N");
+    find_type_by_name<TypeRef>(db.TypeRef, "E1", "N");
+    auto const& S1_ref = find_type_by_name<TypeRef>(db.TypeRef, "S1", "N");
+
+    REQUIRE(db.TypeDef.size() == TYPE_DEF_OFFSET + 3);
+    auto const& combo_def = find_type_by_name<TypeDef>(db.TypeDef, "IComboBox", "N");
+    test_interface_type_properties(combo_def);
+
+    REQUIRE(size(combo_def.PropertyList()) == 1);
+    auto const& property_list = combo_def.PropertyList();
+
+    auto const& property1 = property_list.first[0];
+    REQUIRE(property1.Name() == "property1");
+    REQUIRE(property1.Flags().value == method_attributes_no_flags().value);
+    
+    REQUIRE(db.MethodSemantics.size() == 2);
+    for (size_t i = 0; i < db.MethodSemantics.size(); i++)
+    {
+        auto const& assoc = db.MethodSemantics[i].Association().type();
+        REQUIRE(assoc == HasSemantics::Property);
+        auto const& property_sig = db.MethodSemantics[i].Association().Property().Type();
```

Comment:
`Association().Property()` should actually be equal to the `property1` variable declared above. So you can check the property signature and name outside the loop. All that needs to happen down in this loop is that the entry in the semantics table is equal to the `property1`.